buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionArchiveSetupPlugin.java

```diff
@@ -85,11 +87,16 @@ private void registerAndConfigureDistributionArchivesExtension(Project project)
             project.project(subProjectName, sub -> {
                 sub.getPlugins().apply(BasePlugin.class);
                 sub.getArtifacts().add(DEFAULT_CONFIGURATION_NAME, distributionArchive.getArchiveTask());
-                var extractedConfiguration = sub.getConfigurations().create("extracted");
+                var extractedConfiguration = sub.getConfigurations().create(EXTRACTED_CONFIGURATION_NAME);
                 extractedConfiguration.setCanBeResolved(false);
                 extractedConfiguration.getAttributes().attribute(ARTIFACT_FORMAT, ArtifactTypeDefinition.DIRECTORY_TYPE);
                 sub.getArtifacts().add(EXTRACTED_CONFIGURATION_NAME, distributionArchive.getExpandedDistTask());
-
+                sub.getTasks().register("extractedAssemble", new Action<Task>() {
+                    @Override
+                    public void execute(Task task) {
+                        task.dependsOn(extractedConfiguration.getAllArtifacts().getBuildDependencies());
```

Comment:
Doesn't just depending on the configuration itself work? Why the need to explicitly reference the build dependencies?

Reply:
I think you're right 🤦